micro-rdk/src/common/robot.rs

```diff
@@ -230,7 +242,7 @@ impl LocalRobot {
 
     fn build_resource(
         &mut self,
-        config: &DynamicComponentConfig,
+        config: &mut DynamicComponentConfig,
```

Comment:
it's because I'm using drain to process the collector configs in an act of premature optimization. I'll change it to a normal iterator and then I should be able to change these back